applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -919,6 +929,116 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::ComputePotentia
     }
 }
 
+template <int TDim, int TNumNodes>
+void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::FindUpwindElement(const ProcessInfo& rCurrentProcessInfo)
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+
+    // current element geometry
+    const GeometryType& r_geom = r_this.GetGeometry();
+    const Point current_element_center = r_geom.Center();
+
+    // current element edges
+    const auto edges = r_geom.GenerateEdges();
+
+    // get local coordinates of element center
+    array_1d<double,3> aux_coordinates;
+    r_geom.PointLocalCoordinates(aux_coordinates, current_element_center);
+
+    // outward pointing normals of each edge
+    auto first_edge_normal = edges[0].Normal(aux_coordinates);
```

Comment:
JFYI the normal you retrieve here doesn't have unit magnitude. I think we should normalize the normal instead of the free stream velocity.

Reply:
As we need to compare the different normal components in the free stream direction shouldn't we normalize by the free stream velocity, so that they're all a 'fair' comparison?

Reply:
Maybe I understand it wrong, but as far as I see it, in order to achieve a fair comparison we need a normalized normal, not a normalized free stream velocity. We can discuss it next time